apps/eth_blockchain/lib/eth_blockchain/transaction.ex

```diff
@@ -137,31 +127,73 @@ defmodule EthBlockchain.Transaction do
          adapter,
          pid
        ) do
-    with {:ok, encoded_abi_data} <- ABIEncoder.transfer(to, amount),
-         {:ok, nonce} <- get_transaction_count(%{address: from}, adapter, pid) do
-      gas_limit = Application.get_env(:eth_blockchain, :default_contract_transaction_gas_limit)
-      gas_price = get_gas_price_or_default(attrs)
-
+    with {:ok, meta} <-
+           get_transaction_meta(attrs, :default_contract_transaction_gas_limit, adapter, pid),
+         {:ok, encoded_abi_data} <- ABIEncoder.transfer(to, amount) do
       %__MODULE__{
-        gas_limit: gas_limit,
-        gas_price: gas_price,
-        nonce: int_from_hex(nonce),
         to: from_hex(contract_address),
         data: encoded_abi_data
       }
+      |> Map.merge(meta)
       |> sign_and_hash(from)
       |> send_raw(adapter, pid)
     else
       error -> error
     end
   end
 
-  defp get_gas_price_or_default(%{gas_price: gas_price}), do: gas_price
+  @doc """
+  Submit a contract creation transaction with the given data
+  Returns {:ok, tx_hash, contract_address} if success,
+  {:error, code} || {:error, code, message} otherwise
+  """
+  def create_contract(%{from: from, contract_data: init} = attrs, adapter, pid) do
```

Comment:
I know this is just sending a transaction to create it, but would it make more sense in a `Contract` module?